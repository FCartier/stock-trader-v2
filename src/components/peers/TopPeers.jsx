import React from "react";

import { SectionHeader } from "../reusables/SectionHeader"
import  NoData  from "../reusables/NoData";

const TopPeers = ({toppeers}) => {
  return toppeers.length ? (
    <div>
      {toppeers && (
        <div>
          <SectionHeader title="TOP PEERS"/>
          <div>
            <ul className="top-peers-list">
              {toppeers.length 
              ? (toppeers.map((peer, index) => {
                return (
                  <li className="top-peers-element" key={index}>
                    {peer}
                  </li>
                );
              }))
              : <NoData componentName="top peers"/>}
              
            </ul>
          </div>
        </div>
      )}
    </div>
  ) : null
};


export default TopPeers;
