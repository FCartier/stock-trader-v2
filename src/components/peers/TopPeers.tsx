import * as React from "react";
import { SectionHeader } from "../reusables/SectionHeader";
import { List, Item } from "./styles";

interface ITopPeersProps {
  peers?: string[];
}

const TopPeers: React.SFC<ITopPeersProps> = ({ peers }) => {
  return peers.length ? (
    <div>
      <SectionHeader title="TOP PEERS" />
      <div>
        <List>
          {peers.slice(0, 6).map((peer: string, index: number) => {
            return <Item key={index}>{peer}</Item>;
          })}
        </List>
      </div>
    </div>
  ) : null;
};

export default TopPeers;
