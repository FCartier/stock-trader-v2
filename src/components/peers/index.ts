import PeersContainer from "./peersContainer"

export default PeersContainer