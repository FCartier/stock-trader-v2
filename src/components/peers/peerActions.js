export const PEER_ACTION = "PEER ACTION";


export const peerAction = payload => ({
    type: PEER_ACTION,
    payload
  });