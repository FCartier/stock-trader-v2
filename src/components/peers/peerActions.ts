export const FETCH_PEER_SUCCESS = "FETCH_PEER_SUCCESS";
export type FETCH_PEER_SUCCESS = typeof FETCH_PEER_SUCCESS;
