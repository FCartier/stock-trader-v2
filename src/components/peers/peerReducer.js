import { PEER_ACTION } from "./peerActions"

export const peers = (state = [], action) => {
    switch (action.type) {
      case PEER_ACTION:
        return action.payload;
      default:
        return state;
    }
  };