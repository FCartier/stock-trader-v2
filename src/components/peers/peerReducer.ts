import { Reducer } from "redux";
import { FETCH_PEER_SUCCESS, FETCH_PEER_FAILED } from "./peerActions";
import PeerState from "./types";
import { fetchStatus } from "../reusables/fetchStatus";

const initialState: PeerState = {
  peers: [],
  status: fetchStatus.pending
};

export const peers: Reducer<PeerState> = (state = initialState, action) => {
  switch (action.type) {
    case FETCH_PEER_SUCCESS:
      return { peers: action.payload, status: fetchStatus.success };
    case FETCH_PEER_FAILED:
      return { ...state, status: fetchStatus.failed}
    default:
      return state;
  }
};
