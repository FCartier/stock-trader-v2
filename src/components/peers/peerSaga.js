import { call, put, takeLatest } from "redux-saga/effects";
import { FETCH_PEER_SUCCESS } from "./peerActions";
import { SELECTED_SYMBOL } from "../search/searchActions"
import { api } from "../../utils/apiUtil";

/**** Workers ****/

function* peerWorker({payload}) {
  const toppeers = yield call(api.getPeers, payload);
  yield put({ type: FETCH_PEER_SUCCESS, payload: toppeers });
}

/**** Watchers ****/

export function* peerWatcher() {
  yield takeLatest(SELECTED_SYMBOL, peerWorker);
}
