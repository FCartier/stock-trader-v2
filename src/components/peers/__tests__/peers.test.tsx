import * as React from "react";
import { configure, shallow } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import TopPeers from "../TopPeers";
import { testData } from "../__data__/peers.data";
import { fetchStatus } from "../../reusables/fetchStatus";

configure({ adapter: new Adapter() });

describe("peers component", () => {
  test("peers component should render properly before fetch", () => {
    const component = shallow(<TopPeers  peers={[]} status={fetchStatus.pending} />);
    expect(component).toMatchSnapshot();
  });

  test("peers component should render properly with fetch success", () => {
    const component = shallow(<TopPeers  peers={testData} status={fetchStatus.success} />);
    expect(component).toMatchSnapshot();
  });

  test("peers component should render correctly with fetch failed", () => {
    const component = shallow(<TopPeers  peers={testData} status={fetchStatus.failed} />);
    expect(component).toMatchSnapshot();
  });
});
