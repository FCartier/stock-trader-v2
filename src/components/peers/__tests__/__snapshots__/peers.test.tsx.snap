// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`peers component peers component should render correctly with fetch failed 1`] = `
<div>
  <Component
    title="TOP PEERS"
  />
  <div />
</div>
`;

exports[`peers component peers component should render properly before fetch 1`] = `""`;

exports[`peers component peers component should render properly with fetch success 1`] = `
<div>
  <Component
    title="TOP PEERS"
  />
  <div>
    <div>
      <Styled(ul)>
        <Styled(li)
          key="0"
        >
          AAPL
        </Styled(li)>
        <Styled(li)
          key="1"
        >
          FB
        </Styled(li)>
        <Styled(li)
          key="2"
        >
          GOOG
        </Styled(li)>
        <Styled(li)
          key="3"
        >
          GOOGL
        </Styled(li)>
      </Styled(ul)>
    </div>
  </div>
</div>
`;
